### PR TITLE
[Slim] Improve codebase decouple

### PR DESCRIPTION
**Pull request by ybelenko**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.1.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Logical prepare before unit test generation feature.
Now generated package splitted into ApiControllers, Models and SlimRouter. This provides simple approach for unit testing and code coverage.
Samples omitted in first commit to simplify code review.

cc @jebentier @dkarlovi @mandrean @jfastnacht @ackintosh

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PhpSlimServerCodegen.java [hunk 1]
```diff
@@ -99,6 +87,43 @@ public String getHelp() {
         return "Generates a PHP Slim Framework server library.";
     }
 
+    @Override
+    public void setInvokerPackage(String invokerPackage) {
+        super.setInvokerPackage(invokerPackage);
+        apiPackage = invokerPackage + "\\" + apiDirName;
+        modelPackage = invokerPackage + "\\" + modelDirName;
```
Comment (ackintosh) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PhpSlimServerCodegen.java]:
```diff
+        apiPackage = invokerPackage + "\\" + apiDirName;
+        modelPackage = invokerPackage + "\\" + modelDirName;
```

As the method name `setInvokerPackage()` implies, it should set only `invokerPackage`. 💡

Comment (ybelenko) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PhpSlimServerCodegen.java]: @ackintosh 
I've copied it from PhpSymfony generator.
https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PhpSymfonyServerCodegen.java#L481-L490

Where I should place these changes then?

**Comment by ybelenko** [comment]
Fixed. `setInvokerPackage` overriden method deleted. 
I will try to solve Composer autoload issue in next PRs, for now user can set wrong values(when `modelPackage` and `apiPackage` doesn't start with `invokerPackage` Composer autoloader is broken). Maybe we can just throw a warning to user instead of fixing them when provided values are incorrect.

**Comment by ybelenko** [comment]
### Upgrade note

Current refactoring brings breaking changes. If your application stop working you need to copypaste endpoints implementation from `index.php` to corresponding `ApiControllers`(OpenAPIServer\Api\FakeApi class for instance). That simple code relocation should fix your app.

**Comment by wing328** [comment]
The Shippable failure has been addressed in the master.

**Pull request merged**